src/LoggerAbstractServiceFactory.php

```diff
@@ -63,21 +75,28 @@ public function __invoke(ContainerInterface $container, $requestedName, ?array $
     }
 
     /**
-     * {@inheritDoc}
+     * @param string $name
+     * @param string $requestedName
+     * @return Logger
+     * @throws ContainerException
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
      */
-    public function createServiceWithName(ServiceLocatorInterface $container, $name, $requestedName)
+    public function createServiceWithName(ServiceLocatorInterface $serviceLocator, $name, $requestedName)
     {
-        return $this($container, $requestedName);
+        return $this($serviceLocator, $requestedName);
     }
 
     /**
      * Retrieve configuration for loggers, if any
      *
      * @return array
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
      */
     protected function getConfig(ContainerInterface $services)
     {
-        if ($this->config !== null) {
+        if (isset($this->config)) {
```

Comment:
`isset()` should be reserved for cases where a property may not be declared: for nullable scenarios, please keep `!== null`, which leads to a notice or warning, if a property no longer exists for any reason.

Be very cautious around `isset()` and `empty()` usages: they are more complex than what they look like :)

Reply:
Uninitialized properties after `__construct` are usually considered bugs :)